samples/cellular/modem_shell/src/fota/fota_shell.c

```diff
@@ -13,33 +13,14 @@
 #include "fota.h"
 #include "mosh_print.h"
 
-static const char fota_server_eu[] = "nrf-test-eu.s3.amazonaws.com";
-static const char fota_server_usa[] = "nrf-test-us.s3.amazonaws.com";
-static const char fota_server_jpn[] = "nrf-test-jpn.s3.amazonaws.com";
-static const char fota_server_au[] = "nrf-test-au.s3.amazonaws.com";
-
 static int cmd_fota_download(const struct shell *shell, size_t argc,
 			     char **argv)
 {
 	int err;
-	const char *fota_server;
-
-	if (strcmp(argv[1], "eu") == 0) {
-		fota_server = fota_server_eu;
-	} else if (strcmp(argv[1], "us") == 0) {
-		fota_server = fota_server_usa;
-	} else if (strcmp(argv[1], "jpn") == 0) {
-		fota_server = fota_server_jpn;
-	} else if (strcmp(argv[1], "au") == 0) {
-		fota_server = fota_server_au;
-	} else {
-		mosh_error("FOTA: Unknown server: %s", argv[1]);
-		return -EINVAL;
-	}
 
 	mosh_print("FOTA: Starting download...");
 
-	err = fota_start(fota_server, argv[2]);
```

Comment:
The `fota download` command has been intended for our internal use in (automated) testing and certification. The average user can not really use it, because it's not possible to list the files on our servers.

Changing the syntax of the command would lead to updating all tests and specifications referring to the command. CI is handled easily, but I'm worried about the certification. There are our own field test engineers, but also 3rd parties running tests with mosh, and I don't even know what would be affected.

Unless there's a really good reason to change the command syntax, I would keep it as it is. That said, there could be a new command or an option to download a FOTA update using an URI. That's fine, as long as the current functionality remains unchanged.

Reply:
Reason for changing it was that the API has changed to take the URI instead of server and client separately, so this would avoid the need for an intermediate buffer, in addition to the command being usable for other servers as well. Though, I was not aware of the extent it was used for testing. I've updated the file to keep the old syntax. Though, I wonder if it would be ok to treat the input as an uri if we don't recognize the server?

Reply:
> Though, I wonder if it would be ok to treat the input as an uri if we don't recognize the server?

This was on my mind as well as a possible implementation for the wider support of any server address.